python/tskit/drawing.py

```diff
@@ -551,28 +551,31 @@ def draw_y_axis(
             )
         if self.y_axis:
             y_axis.add(dwg.line((x, rnd(lower)), (x, rnd(upper))))
-            if tick_positions is not None:
-                for pos in tick_positions:
-                    tick = y_axis.add(
-                        dwg.g(
-                            class_="tick",
-                            transform=f"translate({x} {rnd(self.y_transform(pos))})",
-                        )
+            try:
```

Comment:
There's awful lot of logic in this function already. How about 
```python
def check_yticks(ticks: Union[List, Mapping, None]) -> Mapping: # Not sure what the actual types would be here, just e.g.
    if ticks is None:
        return {}
    if isinstance(ticks, Mapping):
       return ticks
   return {pos: "{pos:.2f}" for pos in ticks}
```

The check_yticks function can then be tested independently, and it's not bound up with loads of other, unrelated, logic.

Then later:

```
ticks = check_yticks(ticks)
```

Reply:
Yep, I guess that's possible. I'll work though the logic to see if it hangs together.